### PR TITLE
Initial SBOM support

### PR DESCRIPTION
**Pull request by juli1**
### What and why?

 - **What**: we want to add support to upload SBOM payload to Datadog endpoint.
 - **Why**: we want to offer a new feature to review SBOM

### How?

This PR is the first PR to introduce support for SBOM file upload. It works as follow

1. Check the file is valid
2. Generate the protobuf representation of the data
3. Upload the data to the intake endpoint.

**Important notes**
 - Opposite to existing features, we do not make use of the concurrency as we need to only upload one or two files.
 - We do not use any renderer (at least for now) and tried to keep the code very minimal
 - When running `yarn install`, some dependencies were updated and broke the existing code. We had to add new `resolutions` to `package.json`
 - The command is available as a beta command and we need to enable `DD_BETA_COMMANDS_ENABLED` to `1`

### Review checklist

- [x] Add unit test for all validation functions

### CODE REVIEW

File: src/helpers/tags.ts [hunk 1]
```diff
@@ -97,3 +103,26 @@ export const parseMetrics = (tags: string[]) => {
     return {}
   }
 }
+
+/**
+ * Get SPAN tags get the tags to upload results in CI for the following commands: sarif and sbom.
```
Comment (dastrong) [inline src/helpers/tags.ts:108]:
```suggestion
 * Get the tags to upload results in CI for the following commands: sarif and sbom.
```

File: src/helpers/config.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface DatadogCiConfig {
+  apiKey: string | undefined
+  env: string | undefined
+  envVarTags: string | undefined
```
Comment (dastrong) [inline src/helpers/config.ts:4]:
```suggestion
  apiKey?: string
  env?: string
  envVarTags?: string
```

File: src/helpers/api.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import {DATADOG_SITE_US1} from '../constants'
+import {APIHelper} from '../commands/sarif/interfaces'
+import chalk from 'chalk'
+import {apiConstructor} from '../commands/sarif/api'
```
Comment (dastrong) [inline src/helpers/api.ts:4]:
```suggestion
import chalk from 'chalk'
import {APIHelper} from '../commands/sarif/interfaces'
import {apiConstructor} from '../commands/sarif/api'
import {DATADOG_SITE_US1} from '../constants'
```

File: src/commands/sbom/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+### Environment variables
+
+The following environment variables must be defined
+
+ - `DD_SITE`: the [Datadog site](https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site)
+ - `DD_APP_KEY`: the App Key to use
+ - `DD_API_KEY`: the API key to use
+ - `DD_SERVICE`: the Datadog service you use
```
Comment (dastrong) [inline src/commands/sbom/README.md:22]: This doesn't need to be defined since you can pass in `--service some-service` though?

**Review by dastrong — CHANGES_REQUESTED** [review]

**Comment by juli1** [comment]
rebase not clean, opened a clean pull request here https://github.com/DataDog/datadog-ci/pull/1038

**Pull request closed without merging**